Plugins/Public/base_plugin/FactoryModule.cpp

```diff
@@ -282,5 +332,52 @@ bool FactoryModule::ToggleQueuePaused(bool NewState)
 {
 	bool RememberState = Paused;
 	Paused = NewState;
-	return RememberState;
+	//return true if value changed
+	return RememberState != NewState;
+}
+
+FactoryModule* FactoryModule::FindModuleByProductInProduction(PlayerBase* pb, uint searchedProduct)
+{
+	for (auto& module : pb->modules)
+	{
+		FactoryModule* facModPtr = dynamic_cast<FactoryModule*>(module);
+		if (facModPtr && facModPtr->active_recipe.nickname == searchedProduct)
```

Comment:
wouldn't it be safest to do a IsFactoryModule-like check on these first?

Reply:
If it's not a FactoryModule, the dynamc cast will return a nullptr, for which I already check.